src/main/java/com/cleanroommc/groovyscript/compat/mods/essentialcraft/DemonTradeManager.java

```diff
@@ -0,0 +1,87 @@
+package com.cleanroommc.groovyscript.compat.mods.essentialcraft;
+
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.Example;
+import com.cleanroommc.groovyscript.api.documentation.annotations.MethodDescription;
+import com.cleanroommc.groovyscript.api.documentation.annotations.RegistryDescription;
+import com.cleanroommc.groovyscript.helper.Alias;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import net.minecraft.item.ItemStack;
+import essentialcraft.api.DemonTrade;
+import net.minecraftforge.fml.common.registry.EntityEntry;
+
+@RegistryDescription(category = RegistryDescription.Category.ENTRIES)
+public class DemonTradeManager extends VirtualizedRegistry<DemonTrade> {
+    public DemonTradeManager() {
+        super(Alias.generateOf("DemonTrade"));
+    }
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(DemonTrade::removeTrade);
+        restoreFromBackup().forEach(r -> {
+            DemonTrade.TRADES.add(r);
+            if (r.desiredItem.isEmpty()) {
+                DemonTrade.ALL_MOBS.add(r.entityType);

```

Comment:
i dont believe `ALL_MOBS` is handled properly in all removal cases - consider instead using `afterScriptLoad`, such as in `Precipitator`

Reply:
Is this because of `removeAll` or am I missing anything else? I don't think anything else can remove the entity type

Reply:
removeAll works, but `remove(IIngredient)` and `remove(EntityEntry)` dont

Reply:
```java
    public DemonTrade(EntityEntry e) {
        this.desiredItem = ItemStack.EMPTY;
        this.entityType = e;
        ALL_MOBS.add(e);
        TRADES.add(this);
    }

    public static void removeTrade(DemonTrade tra) {
        if (tra.entityType != null) {
            ALL_MOBS.remove(tra.entityType);
        }

        TRADES.remove(tra);
    }
```
So theoretically it should work (although the check for `r.entityType != null` is more paradigmatic). I guess something prevents it from working then, huh. I'll test it further after my thesis defence (so in a few days).